product-service/src/main/java/com/extremecoder/productservice/utils/ResponseBuilder.java

```diff
@@ -0,0 +1,80 @@
+package com.extremecoder.productservice.utils;
+
+import com.extremecoder.productservice.dto.ErrorResponseDto;
+import com.extremecoder.productservice.dto.Response;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ResponseBuilder {
+    private ResponseBuilder() {
+    }
+
+    private static List<ErrorResponseDto> getCustomError(BindingResult result) {
+        List<ErrorResponseDto> dtoList = new ArrayList<>();
+        result.getFieldErrors().forEach(fieldError -> {
+            ErrorResponseDto errorResponseDto = ErrorResponseDto.builder()
+                    .field(fieldError.getField())
+                    .message(fieldError.getDefaultMessage())
+                    .build();
+            dtoList.add(errorResponseDto);
+        });
+        return dtoList;
+    }
+
+    public static Response getFailureResponse(BindingResult result, String message) {
+        return Response.builder()
```

Comment:
Instead of writing the line 29 to 35 we could directly write:
 `return getFailureResponse(HTTPStatus.BAD_REQUEST.value(), message).errors(getCustomError(result)).build()`